orc8r/cloud/go/obsidian/tests/unit_test_utils.go

```diff
@@ -86,14 +87,18 @@ func RunUnitTest(t *testing.T, e *echo.Echo, test Test) {
 		} else {
 			assert.Contains(t, handlerErr.Error(), test.ExpectedErrorSubstring)
 		}
-	} else {
-		if assert.NoError(t, handlerErr) && test.ExpectedResult != nil {
+	} else if test.ExpectedResult != nil {
+		if assert.NoError(t, handlerErr) {
 			expectedBytes, err := test.ExpectedResult.MarshalBinary()
 			if assert.NoError(t, err) {
 				// Convert to string for more readable assert failure messages
 				assert.Equal(t, string(expectedBytes)+"\n", recorder.Body.String())
 			}
 		}
+	} else if test.ExpectedString != "" {
```

Comment:
Do we have to be concerned about the case in which a future test case may actually want to test that the expected string is the empty string?